app/src/main/kotlin/compiler/common/reference_collections/ReferenceHashMap.kt

```diff
@@ -37,7 +37,7 @@ fun <K, V> combineReferenceMaps(maps: List<ReferenceMap<K, V>>): ReferenceMap<K,
     return combinedMaps
 }
 
-fun <K, V> referenceMapOf(pairs: List<Pair<K, V>>): ReferenceMap<K, V> {
+fun <K, V> referenceHashMapOf(pairs: List<Pair<K, V>>): ReferenceHashMap<K, V> {
```

Comment:
Since we have this function, it would be a good idea to replace usages of `ReferenceHashMap()` constructor with it in the whole code (for the sake of consistency). Also, maybe we should change `referenceSetOf()` into `referenceHashSetOf()` as well.